Error with -pedandic -Werror when including caml/domain_state.h

```
#=== ERROR while compiling bytearray.1.0.1 ====================================#
# context              2.1.0~beta4 | linux/x86_64 | ocaml-variants.4.10.0+multicore+no-effect-syntax | file:///home/opam/opam-repository
# path                 ~/.opam/4.10.0+multicore+no-effect-syntax/.opam-switch/build/bytearray.1.0.1
# command              ~/.opam/4.10.0+multicore+no-effect-syntax/bin/dune build -p bytearray -j 55
# exit-code            1
# env-file             ~/.opam/log/bytearray-7-629740.env
# output-file          ~/.opam/log/bytearray-7-629740.out
### output ###
#          gcc src/bytearray_stubs.o (exit 1)
# (cd _build/default/src && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -g -O2 -fPIC -Wall -pedantic -Werror -Wno-long-long -g -I /home/opam/.opam/4.10.0+multicore+no-effect-syntax/lib/ocaml -o bytearray_stubs.o -c bytearray_stubs.c)
# In file included from /home/opam/.opam/4.10.0+multicore+no-effect-syntax/lib/ocaml/caml/mlvalues.h:31,
#                  from /home/opam/.opam/4.10.0+multicore+no-effect-syntax/lib/ocaml/caml/intext.h:25,
#                  from bytearray_stubs.c:6:
# /home/opam/.opam/4.10.0+multicore+no-effect-syntax/lib/ocaml/caml/domain_state.h:63:1: error: embedding a directive within macro arguments is not portable [-Werror]
#    63 | #ifndef NATIVE_CODE
#       | ^
# /home/opam/.opam/4.10.0+multicore+no-effect-syntax/lib/ocaml/caml/domain_state.h:65:1: error: embedding a directive within macro arguments is not portable [-Werror]
#    65 | #endif
#       | ^
# cc1: all warnings being treated as errors
```

--- a/runtime/caml/domain_state.h
+++ b/runtime/caml/domain_state.h
@@ -57,13 +57,16 @@ enum {
 
 /* Check that the structure was laid out without padding,
    since the runtime assumes this in computing offsets */
+#ifdef NATIVE_CODE
+CAML_STATIC_ASSERT(
+  sizeof(caml_domain_state) == Domain_state_num_fields * 8);
+#else
 CAML_STATIC_ASSERT(
   sizeof(caml_domain_state) ==
    (Domain_state_num_fields
-#ifndef NATIVE_CODE
      + Byte_domain_state_num_fields
-#endif
    ) * 8);
+#endif
 
 #ifdef __APPLE__
   CAMLextern pthread_key_t caml_domain_state_key;

